test/jdk/javax/swing/JPopupMenu/NestedFocusablePopupTest.java

```diff
@@ -64,71 +69,87 @@ public static void main(String[] args) throws Exception {
 
         try {
             SwingUtilities.invokeAndWait(NestedFocusablePopupTest::initAndShowGui);
-            robot.waitForIdle();
-            robot.delay(500);
             test0();
-
-            robot.waitForIdle();
-            robot.delay(500);
             test1();
         } finally {
             SwingUtilities.invokeAndWait(frame::dispose);
         }
     }
 
+    public static void waitTillShown(final Component component, long msTimeout)
+            throws InterruptedException, TimeoutException {
+        long startTime = System.currentTimeMillis();
+
+        while (true) {
+            try {
+                Thread.sleep(50);
+                component.getLocationOnScreen();
+                break;
+            } catch (IllegalComponentStateException e) {
+                if (System.currentTimeMillis() - startTime > msTimeout) {
+                    throw new TimeoutException("Component not shown within the specified timeout");
+                }
+            }
+        }
+    }
+
+    static Rectangle waitAndGetOnScreenBoundsOnEDT(Component component) throws Exception {
```

Comment:
```suggestion
    static Rectangle waitAndGetOnScreenBoundsOnEDT(Component component)
            throws InterruptedException, TimeoutException, ExecutionException {
```
For consistency with `waitTillShown` which lists exceptions.

Reply:
Updated